common/transport/amqp/src/amqp_message.ts

```diff
@@ -120,12 +120,23 @@ export class AmqpMessage {
         if (!amqpMessage.application_properties) {
           ensureApplicationPropertiesCreated();
         }
+        const DT_SUBJECT = 'dt-subject';
         for (let index = 0; index < propsCount; index++) {
           const item = props.getItem(index);
           if (!!item) {
-            /*Codes_SRS_NODE_IOTHUB_AMQPMSG_16_013: [If one of the property key is `IoThub-status`, this property is reserved and shall be forced to an `int` `rhea` type.]*/
-            const val = (item.key === 'IoThub-status') ? rheaTypes.wrap_int(parseInt(item.value)) : item.value;
-            amqpMessage.application_properties[item.key] = val;
+            if (item.key === DT_SUBJECT) {
```

Comment:
>if (item.key === DT_SUBJECT) { [](http://example.com/codeflow?start=12&length=30)

I don't know if this matters, but `ensureApplicationPropertiesCreated()` gets called even if there is a single `dt-subject` properties (and thus no application properties.) Do you want to move the call to `ensureApplicationPropertiesCreated` down below where it actually sets the property?

Reply:
Done